ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -114,7 +115,8 @@ private static Map<String, Object> processResponse(SearchHit hit) {
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
         try {
             String input = parameters.get(INPUT_FIELD);
-            JsonObject jsonObject = StringUtils.gson.fromJson(input, JsonObject.class);
+            Gson gson = new GsonBuilder().serializeSpecialFloatingPointValues().create();
```

Comment:
there is gson in StringUtils

Reply:
can we at least have this as a static variable?

```
private static final Gson GSON = new GsonBuilder()
    .serializeSpecialFloatingPointValues()
    .create();
```